doc/modules/mod_http_upload.md

```diff
@@ -7,23 +7,93 @@ Currently, the module supports only the [S3][s3] backend using [AWS Signature Ve
 
 ### Options
 
-* **iqdisc** (default: `one_queue`)
-* **host** (string, default: `"upload.@HOST@"`): Subdomain for the upload service to reside under. `@HOST@` is replaced with each served domain.
-* **backend** (atom, default: `s3`) - Backend to use for generating slots. Currently only `s3` can be used.
-* **expiration_time** (integer, default: `60`) - Duration (in seconds) after which the generated `PUT` URL will become invalid.
-* **token_bytes** (integer, default: `32`) - Number of random bytes of a token that will be used in a generated URL. 
- The text representation of the token will be twice as long as the number of bytes, e.g. for the default value the token in the URL will be 64 characters long.
-* **max_file_size** (integer, default: 10485760 (10 MB)) - Maximum file size (in bytes) accepted by the module. Disabled if set to `undefined`.
-* **s3** (list, default: unset) - Options specific to [S3][s3] backend.
+#### `modules.mod_http_upload.iqdisc.type`
+* **Syntax:** string, one of `"one_queue"`, `"no_queue"`, `"queues"`, `"parallel"`
+* **Default:** `"one_queue"`
+
+Strategy to handle incoming stanzas. For details, please refer to
+[IQ processing policies](../../advanced-configuration/Modules/#iq-processing-policies).
+
+#### `modules.mod_http_upload.host`
+* **Syntax:** string
+* **Default:** `"upload.@HOST@"`
+* **Example:** `host = "upload.@HOST@"`
+
+Subdomain for the upload service to reside under. `@HOST@` is replaced with each served domain.
+
+#### `modules.mod_http_upload.backend`
+* **Syntax:** non-empty string
+* **Default:** `"s3"`
+* **Example:** `backend = "s3"`
+
+Backend to use for generating slots. Currently only `"s3"` can be used.
+
+#### `modules.mod_http_upload.expiration_time`
+* **Syntax:** non-negative integer
+* **Default:** `60`
+* **Example:** `expiration_time = 120`
+
+Duration (in seconds) after which the generated `PUT` URL will become invalid.
+
+#### `modules.mod_http_upload.token_bytes`
+* **Syntax:** positive integer
+* **Default:** `32`
+* **Example:** `token_bytes = 32`
+
+Number of random bytes of a token that will be used in a generated URL. 
+The text representation of the token will be twice as long as the number of bytes, e.g. for the default value the token in the URL will be 64 characters long.
+
+#### `modules.mod_http_upload.max_file_size`
+* **Syntax:** non-negative integer or the string `"undefined"`
```

Comment:
Don't think `"undefined"` will pass the validation

Reply:
Just skip 'undefined' here and specify the default as 'not set - no size limit'